tree/tree/inc/ROOT/InternalTreeUtils.hxx

```diff
@@ -105,6 +106,8 @@ public:
    ClassDef(RNoCleanupNotifier, 0);
 };
 
+std::unique_ptr<TChain> MakeChainForMT(const std::string &name = "", const std::string &title = "");
```

Comment:
Minor question on the signature.  Since inside the argument as used only as C-strings, would it make sense to delegate to the caller to call `c_str()` in the (unlikely?) case they use an std::string to build it?  (otherwise, all the calls that do not get an std::string need to create one, in particular when the arguments are not passed, need to create 2 std::string)  [and/or is the cost of this creation for sure negligible?

Reply:
Only new ROOT code will call this, so I expect it to use `std::strings` rather than `const char *`. The idea was to write the `c_str()` calls once here rather than many times at each call site.

Reply:
> so I expect it to use std::strings

Except the use in this PR :) which does not pass any arguements (i.e. create empty std::string to pass '""' through).

Reply:
One use uses the default arguments, the other passes a std::string (and removes a c_str() call from client code).

I don't see any advantage in taking two `const char *` instead.

Reply:
> I don't see any advantage in taking two const char * instead.

It hinges on the actual cost (these days) of creating an empty std::string.  Since it is internal, taking an std::string is merely a developer convenience (avoiding typing `.c_str()` at a few call site) at the (to be determine) cost of creating the empty std::string.

Reply:
that cost is very little (all STL implementations have a small string optimization so an empty string does not do any malloc) and negligible w.r.t. creating the TChain object anyway.